src/main/kotlin/com/mineinabyss/looty/features/recipes/ItemRecipeSystem.kt

```diff
@@ -52,4 +51,11 @@ class RecipeDiscoverySystem(
     fun PlayerJoinEvent.showRecipesOnJoin() {
         player.discoverRecipes(discoveredRecipes)
     }
+
+    @EventHandler(priority = EventPriority.HIGHEST)
+    fun PrepareItemCraftEvent.onCraftWithCustomItem() {
+        if (inventory.matrix.mapNotNull {
```

Comment:
Use asSequence here in the beginning so that we dont first map all elements, then check for components one by one. A sequence will skip getting itemMeta for subsequent items.